src/nvim/search.c

```diff
@@ -710,23 +710,15 @@ int searchit(
                 /* for empty match: advance one char */
                 if (matchcol == matchpos.col
                     && ptr[matchcol] != NUL) {
-                  if (has_mbyte)
-                    matchcol +=
-                      (*mb_ptr2len)(ptr + matchcol);
-                  else
-                    ++matchcol;
+                  matchcol += (*mb_ptr2len)(ptr + matchcol);
```

Comment:
Please use `mb_ptr2len(` instead of `(*mb_ptr2len)(`, usage is already mixed and it is going to be a function soon (and IMHO the former should been used anyway, that mbyte uses function pointers are an implemention detail of the mbyte module that shouldn't had to leak all over the codebase...)